### ISSUE TITLE
Maybe_ generate wrong (non intuitive) sql query

### ISSUE THREAD

**Issue by sashasashasasha151** [haskell-beam/beam#524]
I have left and right table, where right table contains nullable fields \
I join them with `leftJoin_` and try to create `filter_` which has such condition:
```haskell
maybe_ nothing_ nullableFieldFromRightTable rt /=. just_ (val_ 1)
```

So, I if `rt` is null just return `nothing` and if `rt` is not null return `nullableFieldFromRightTable` which will return `nothing` on `just value` for me\
I am using at as beam doen't support to check nested maybes\
Then I just check if it is different from `just 1`\
This code compiles but I get invalid result, so, here is why:

```sql
CASE
  WHEN ((((("t1"."someField1") IS NOT NULL) AND (("t1"."someField2") IS NOT NULL)) AND
    (("t1"."nullableFieldFromRightTable") IS NOT NULL)) AND <all_other_fields_from_table> IS NOT NULL) 
    THEN "t1"."nullableFieldFromRightTable"
  ELSE null END) IS DISTINCT FROM 1)
```

So, it generates strange query which doesn't the meaning of the beam query

**Comment by kmicklas** [comment]
Can you explain how this differs from what you expected? Is it that `maybe_` takes the "just" case only if all columns from `rt` are non-null? This does seem to cause confusion, for example see #442.

**Comment by sashasashasasha151** [comment]
> Is it that maybe_ takes the "just" case only if all columns from rt are non-null?

Yes, I already checked that the root of the problem is in SQL but not in beam. But it will be useful to see it in documentation. I just catched this issue easily because it quickly ruined my project and I knew where to search. But it may cause really big problems for somebody. \
Or maybe it is possible to create a new method like `strictMaybe` of something like this that will return `True` if at least one element in joined table is not null. Yes it will not work for tables with all nullable columns and I am not sure that it is possible to solve it somehow but it is still better than nothing.
